drivers/rkt/driver.go

```diff
@@ -0,0 +1,1007 @@
+	}
+	uuid := strings.TrimSpace(outBuf.String())
+	d.logger.Debug("taskConfig prepared", "pod", img, "task_name", cfg.Name, "uuid", uuid)
+	runArgs = append(runArgs, uuid)
+
+	// The taskConfig's environment is set via --set-env flags above, but the rkt
+	// command itself needs an environment with PATH set to find iptables.
+
+	// TODO (preetha) need to figure out how to pass env.blacklist from client config
+	eb := env.NewEmptyBuilder()
+	filter := strings.Split(config.DefaultEnvBlacklist, ",")
+	rktEnv := eb.SetHostEnvvars(filter).Build()
+
+	// Enable ResourceLimits to place the executor in a parent cgroup of
+	// the rkt container. This allows stats collection via the executor to
+	// work just like it does for exec.
+	execCmd := &executor.ExecCommand{
+		Cmd:            absPath,
+		Args:           runArgs,
+		ResourceLimits: true,
+		Resources: &executor.Resources{
+			CPU:      int(cfg.Resources.LinuxResources.CPUShares),
+			MemoryMB: int(drivers.BytesToMB(cfg.Resources.LinuxResources.MemoryLimitBytes)),
+			DiskMB:   cfg.Resources.NomadResources.DiskMB,
+		},
+		Env:        cfg.EnvList(),
+		TaskDir:    cfg.TaskDir().Dir,
+		StdoutPath: cfg.StdoutPath,
+		StderrPath: cfg.StderrPath,
+	}
+	ps, err := execImpl.Launch(execCmd)
+	if err != nil {
+		pluginClient.Kill()
+		return nil, nil, err
+	}
+
+	d.logger.Debug("started taskConfig", "aci", img, "uuid", uuid, "task_name", cfg.Name, "args", runArgs)
+	h := &rktTaskHandle{
+		exec:         execImpl,
+		env:          rktEnv,
+		pid:          ps.Pid,
+		uuid:         uuid,
+		pluginClient: pluginClient,
+		taskConfig:   cfg,
+		procState:    drivers.TaskStateRunning,
+		startedAt:    time.Now().Round(time.Millisecond),
+		logger:       d.logger,
+	}
+
+	rktDriverState := RktTaskState{
+		ReattachConfig: utils.ReattachConfigFromGoPlugin(pluginClient.ReattachConfig()),
+		Pid:            ps.Pid,
+		TaskConfig:     cfg,
+		StartedAt:      h.startedAt,
+		UUID:           uuid,
+	}
+
+	if err := handle.SetDriverState(&rktDriverState); err != nil {
+		d.logger.Error("failed to start task, error setting driver state", "error", err)
+		execImpl.Shutdown("", 0)
+		pluginClient.Kill()
+		return nil, nil, fmt.Errorf("failed to set driver state: %v", err)
+	}
+
+	d.tasks.Set(cfg.ID, h)
+	go h.run()
+
+	// Do not attempt to retrieve driver network if one won't exist:
+	//  - "host" means the container itself has no networking metadata
+	//  - "none" means no network is configured
+	// https://coreos.com/rkt/docs/latest/networking/overview.html#no-loopback-only-networking
+	var driverNetwork *cstructs.DriverNetwork
+	if network != "host" && network != "none" {
+		d.logger.Debug("retrieving network information for pod", "pod", img, "UUID", uuid, "task", cfg.Name)
+		driverNetwork, err = rktGetDriverNetwork(uuid, mergedPortMap, d.logger)
+		if err != nil && !pluginClient.Exited() {
+			d.logger.Warn("network status retrieval for pod failed", "pod", img, "UUID", uuid, "task", cfg.Name, "error", err)
```

Comment:
"task" -> "task_name"

(I kind of regret going with `task_name` because it's needlessly long, but we should keep it consistent.)